tmail-backend/combined-identity/src/main/java/com/linagora/tmail/combined/identity/CombinedUserDAO.java

```diff
@@ -0,0 +1,73 @@
+package com.linagora.tmail.combined.identity;
+
+import java.util.Iterator;
+import java.util.Optional;
+
+import javax.inject.Inject;
+
+import org.apache.james.core.Username;
+import org.apache.james.user.api.UsersRepositoryException;
+import org.apache.james.user.api.model.User;
+import org.apache.james.user.cassandra.CassandraUsersDAO;
+import org.apache.james.user.ldap.ReadOnlyLDAPUsersDAO;
+import org.apache.james.user.lib.UsersDAO;
+
+public class CombinedUserDAO implements UsersDAO {
+
+    private final ReadOnlyLDAPUsersDAO readOnlyLDAPUsersDAO;
+    private final CassandraUsersDAO cassandraUsersDAO;
+
+    @Inject
+    public CombinedUserDAO(ReadOnlyLDAPUsersDAO readOnlyLDAPUsersDAO,
+                           CassandraUsersDAO cassandraUsersDAO) {
+        this.readOnlyLDAPUsersDAO = readOnlyLDAPUsersDAO;
+        this.cassandraUsersDAO = cassandraUsersDAO;
+    }
+
+    @Override
+    public void addUser(Username username, String password) throws UsersRepositoryException {
+        if (readOnlyLDAPUsersDAO.contains(username)) {
+            cassandraUsersDAO.addUser(username, password);
```

Comment:
I don't think we should store the user's password in Cassandra?

Reply:
Honnestly I am reluctant to introduce "yet another table" just to have names.

I think we can accept having dummy passwords here.Specified by the caller. So James is ignorant of them.

Reply:
Yes that was my point, should be dummy, but wanted to be sure! If it's handled by the caller I'm fine then :)